src/jit/lower.cpp

```diff
@@ -462,6 +462,17 @@ void Lowering::DecomposeNode(GenTreePtr* pTree, Compiler::fgWalkData* data)
             hiOp2->gtNext      = hiResult;
             hiResult->gtPrev   = hiOp2;
 
+            if (oper == GT_ADD || oper == GT_SUB)
+            {
+                if (loResult->gtOverflow())
+                {
+                    hiResult->gtFlags |= GTF_OVERFLOW;
```

Comment:
It appears we don't have any tests that test this. I've run the suite with clearing the bit and without and both pass the tests.